go/cmd/server/logging/logging.go

```diff
@@ -0,0 +1,391 @@
+			Name: entityName,
+			Type: arrowType,
+		})
+		columns = append(columns, valArrowArray)
+	}
+
+	for _, featureName := range fcoSchema.Features {
+
+		protoArr := columnNameToProtoValueArray[featureName]
+		if len(protoArr) == 0 {
+			break
+		}
+		arrowArray, err := gotypes.ProtoValuesToArrowArray(protoArr, arrowMemory, len(columnNameToProtoValueArray))
+		if err != nil {
+			return nil, err
+		}
+
+		arrowType, err := gotypes.ValueTypeEnumToArrowType(fcoSchema.FeaturesTypes[featureName])
+
+		if err != nil {
+			return nil, err
+		}
+		fields = append(fields, arrow.Field{
+			Name: featureName,
+			Type: arrowType,
+		})
+		columns = append(columns, arrowArray)
+	}
+	fields = append(fields, arrow.Field{
+		Name: "RequestId",
+		Type: &arrow.StringType{},
+	})
+
+	columns = append(columns, strBuilder.NewArray())
+	schema := arrow.NewSchema(
+		fields,
+		nil,
+	)
+
+	result := array.Record(array.NewRecord(schema, columns, int64(len(memoryBuffer.logs))))
+
+	tbl := array.NewTableFromRecords(schema, []array.Record{result})
+	return array.Table(tbl), nil
+}
+
+type Schema struct {
+	Entities      []string
+	Features      []string
+	EntityTypes   map[string]types.ValueType_Enum
+	FeaturesTypes map[string]types.ValueType_Enum
+}
+
+func GetSchemaFromFeatureService(featureService *model.FeatureService, entities []*model.Entity, entityMap map[string]*model.Entity, featureViews []*model.FeatureView, onDemandFeatureViews []*model.OnDemandFeatureView) (*Schema, error) {
+	fvs := make(map[string]*model.FeatureView)
+	odFvs := make(map[string]*model.OnDemandFeatureView)
+
+	joinKeys := make([]string, 0)
+	// All joinkeys in the featureService are put in here
+	joinKeysSet := make(map[string]interface{})
+	entityJoinKeyToType := make(map[string]types.ValueType_Enum)
+
+	for _, featureView := range featureViews {
+		fvs[featureView.Base.Name] = featureView
+	}
+
+	for _, onDemandFeatureView := range onDemandFeatureViews {
+		odFvs[onDemandFeatureView.Base.Name] = onDemandFeatureView
+	}
+
+	allFeatureTypes := make(map[string]types.ValueType_Enum)
+	features := make([]string, 0)
+	for _, featureProjection := range featureService.Projections {
+		// Create copies of FeatureView that may contains the same *FeatureView but
+		// each differentiated by a *FeatureViewProjection
+		featureViewName := featureProjection.Name
+		if fv, ok := fvs[featureViewName]; ok {
+			for _, f := range featureProjection.Features {
+				features = append(features, GetFullFeatureName(featureViewName, f.Name))
+				allFeatureTypes[GetFullFeatureName(featureViewName, f.Name)] = f.Dtype
+			}
+			for entityName := range fv.Entities {
+				entity := entityMap[entityName]
+				if joinKeyAlias, ok := featureProjection.JoinKeyMap[entity.JoinKey]; ok {
+					joinKeysSet[joinKeyAlias] = nil
+				} else {
+					joinKeysSet[entity.JoinKey] = nil
+				}
+			}
+		} else if odfv, ok := odFvs[featureViewName]; ok {
+			for _, f := range odfv.Base.Features {
```

Comment:
we need to use projection here and not all features (similarly to regular feature views)

Reply:
Actually already fixed.